Hakuba/Source/Section/SectionIndexType.swift

```diff
@@ -0,0 +1,19 @@
+//
+//  SectionIndexType.swift
+//  Example
+//
+//  Created by Keeeeen on 2018/08/09.
+//
+
+import Foundation
+
+public protocol SectionIndexType {
```

Comment:
It may makes more sense like this: 
```swift
public protocol SectionIndexType: RawRepresentable where Self.RawValue == Int {
    static var count: Int { get }
}
```

And, In Swift4.2, we can use CaseIterable in future.

Reply:
> And, In Swift4.2, we can use CaseIterable in future.

YES! I can not wait.